### PR TITLE
[JS] Topic: JSX

### PR DESCRIPTION
**Pull request by SakshiMishra1**
<hr>

## Description 📜

Wrote a tutorial on JSX.
Fix issue #3294

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [x] JavaScript
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by SakshiMishra1** [comment]
@paramsiddharth sir please review.

**Comment by SakshiMishra1** [comment]
issue #3294

File: JavaScript/Topics/Flavours/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# JSX:-
+
+JSX (JavaScript Syntax Extension and occasionally referred as JavaScript XML) is an extension to the JavaScript language syntax[1] which provides a way to structure component rendering using syntax familiar to many developers. It is similar in appearance to HTML.
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:3]:
```suggestion
JSX **(JavaScript Syntax Extension and occasionally referred to as JavaScript XML)** is an extension to the JavaScript language syntax which provides a way to structure component rendering using syntax familiar to many developers. It is similar in appearance to HTML.
```

File: JavaScript/Topics/Flavours/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+JSX converts HTML tags into react elements.
+
+You are not required to use JSX, but JSX makes it easier to write React applications.
+
+Here are two examples. The first uses JSX and the second does not:
+
+Example 1
+**JSX:**
+
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:27]: Write like this:
```jsx
const myelement = <h1>I Love JSX!</h1>;
< ReactDOM.render(myelement, document.getElementById('root'));
```

```
```jsx
const myelement = <h1>I Love JSX!</h1>;
< ReactDOM.render(myelement, document.getElementById('root'));
`` `
```

File: JavaScript/Topics/Flavours/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+## Attributes
+
+JSX provides a range of element attributes designed to mirror those provided by HTML. Custom attributes can also be passed to the component.[4] All attributes will be received by the component as props.
+
+## JavaScript expressions
+
+JavaScript expressions (but not statements) can be used inside JSX with curly brackets {}:
+
+    <h1>{10+1}</h1>
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:52]: Write like this for every code block if jsx:
```jsx
<h1> { 10 + 1 } </h1>
```
```suggestion
    ```jsx
    <h1> { 10 + 1 } </h1>```
```

File: JavaScript/Topics/Flavours/README.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+The example above will render:
+
+    <h1>11</h1>
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:56]:
```suggestion
    ```html
    <h1>11</h1>```
```

**Review by dakshsethi — CHANGES_REQUESTED** [review]
Change all of your code blocks into snippets using ```

**Comment by SakshiMishra1** [comment]
@dakshsethi Made the changes sir

File: JavaScript/Topics/Flavours/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+## Attributes
+
+JSX provides a range of element attributes designed to mirror those provided by HTML. Custom attributes can also be passed to the component.[4] All attributes will be received by the component as props.
+
+## JavaScript expressions
+
+JavaScript expressions (but not statements) can be used inside JSX with curly brackets {}:
+
+```<h1>{10+1}</h1>```
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:53]: Write like this:
```
```jsx
<h1>{10+1}</h1>```

File: JavaScript/Topics/Flavours/README.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+The example above will render:
+
+```<h1>11</h1>```
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:57]: Write like this here:
```
```html
<h1>11</h1>```

File: JavaScript/Topics/Flavours/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+  const i = 1;
+
+  return (
+      <div>
+        <h1>{ i === 1 ? 'true' : 'false' }</h1>
+      </div>
+  );
+}
+```
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:74]: Write like this:
```
```jsx
const App = () => {

  const i = 1;

  return (
      <div>
        <h1>{ i === 1 ? 'true' : 'false' }</h1>
      </div>
  );
}```

File: JavaScript/Topics/Flavours/README.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+The above will render:
+
+```
+<div>
+  <h1>true</h1>
+</div>
+```
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:82]: Write like this:
```
```html
<div>
  <h1>true</h1>
</div>```

File: JavaScript/Topics/Flavours/README.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+          /* 'key' is used by react to keep track of list items and their changes */
+          /* Each 'key' must be unique */
+          <div key={"section-" + n}>
+             Section {n} {i === 0 && <span>(first)</span>}
+          </div>
+        ))}
+       </div>
+   );
+}
+```
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:103]: Write like this here:
```
```jsx
const App = () => {

   const sections = [1, 2, 3];

   return (
      <div>
        {sections.map((n,i) => (
          /* 'key' is used by react to keep track of list items and their changes */
          /* Each 'key' must be unique */
          <div key={"section-" + n}>
             Section {n} {i === 0 && <span>(first)</span>}
          </div>
        ))}
       </div>
   );
}```

File: JavaScript/Topics/Flavours/README.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+The above will render:
+
+```
+<div>
+    <div>Section 1<span>(first)</span></div>
+    <div>Section 2</div>
+    <div>Section 3</div>
+</div>
+```
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:112]:
```
```html
<div>
    <div>Section 1<span>(first)</span></div>
    <div>Section 2</div>
    <div>Section 3</div>
</div>
`` `

File: JavaScript/Topics/Flavours/README.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+JSX allows us to write HTML elements in JavaScript and place them in the DOM without any _createElement()_  and/or _appendChild()_ methods.
+
+JSX converts HTML tags into react elements.
+
+You are not required to use JSX, but JSX makes it easier to write React applications.
+
+Here are two examples. The first uses JSX and the second does not:
+
+Example 1
+```**JSX:**
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:26]: Dont make it bold!!
```
```jsx 
```
Write that...

File: JavaScript/Topics/Flavours/README.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+  const myelement = <h1>I Love JSX!</h1>;
+
+  ReactDOM.render(myelement, document.getElementById('root'));
+```
+
+Example 2
+```**Without JSX:**
```
Comment (dakshsethi) [inline JavaScript/Topics/Flavours/README.md:34]: Make this jsx only
```
```jsx
```
Don't use bold
Mention in a comment that this is without jsx

**Review by dakshsethi — CHANGES_REQUESTED** [review]
Please do all of these changes in a single commit!

**Comment by s4-2** [comment]
@SakshiMishra1  Please write Fix issue #3294 in the documentation section and do check  the required checkbox block:)

File: JavaScript/Topics/Flavours/README.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+<div>
+    <div>Section 1<span>(first)</span></div>
+    <div>Section 2</div>
+    <div>Section 3</div>
+</div>
+```
+
+*Code written in JSX requires conversion with a tool such as Babel before it can be understood by web browsers.*
+
+**The merits of JSX in four bullet points:**
```
Comment (s4-2) [inline JavaScript/Topics/Flavours/README.md:116]: @SakshiMishra1  Just write **Merits of JSX :**

**Comment by SakshiMishra1** [comment]
Made the changes.
Please review @dakshsethi @s4-2 @paramsiddharth

**Review by s4-2 — APPROVED** [review]
Looks good to me. 
@SakshiMishra1 well done:)

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

**Comment by SakshiMishra1** [comment]
Is the PR ready to get merged??
@dakshsethi @paramsiddharth @s4-2

**Comment by paramsiddharth** [comment]
@SakshiMishra1 I'll be relocating the README to Flavours/JSX/README.md after the merge. 😊 Rest looks amazing! The explanation is fantastic. 💕 Good job!

**Review by paramsiddharth — APPROVED** [review]

**Comment by SakshiMishra1** [comment]
> @SakshiMishra1 I'll be relocating the README to Flavours/JSX/README.md after the merge. 😊 Rest looks amazing! The explanation is fantastic. 💕 Good job!

Thank you sir❤

**Pull request merged**